process/scToProtocol/stakingToPeer.go

```diff
@@ -187,15 +188,19 @@ func (stp *stakingToPeer) updatePeerState(
 		return err
 	}
 
-	isValidator := account.GetList() == string(core.EligibleList) || account.GetList() == string(core.WaitingList)
-	unJailForInactive := len(account.GetBLSPublicKey()) > 0 && stakingData.UnJailedNonce == nonce && !isValidator
-	if unJailForInactive {
+	isUnJailForInactive := len(account.GetBLSPublicKey()) > 0 && !stakingData.Staked &&
+		stakingData.UnJailedNonce == nonce && account.GetList() == string(core.JailedList)
+	if isUnJailForInactive {
 		account.SetListAndIndex(account.GetShardId(), string(core.InactiveList), uint32(stakingData.UnJailedNonce))
 		account.SetTempRating(stp.jailRating)
 
 		return stp.peerState.SaveAccount(account)
 	}
 
+	if stakingData.StakedNonce == math.MaxUint64 {
```

Comment:
Could be this condition true when !stakingData.Staked ?

Reply:
Yes. After unStake.

Reply:
ok